components/sql/src/main/scala/pl/touk/nussknacker/sql/service/DatabaseQueryEnricher.scala

```diff
@@ -167,14 +171,13 @@ class DatabaseQueryEnricher(val dbPoolConfig: DBPoolConfig, val dbMetaDataProvid
     }
   }
 
-  // SyntaxErrorException should have message meaningful for users, for others (connection problem?) it's better
-  // to return generic message and log details...
   private def messageFromSQLException(query: String, sqlException: SQLException): String = sqlException match {
     case e: SQLSyntaxErrorException =>
       e.getMessage
-    case e =>
+    case e => // For some users full error msg can be significant. Allow to choose full error message in config
```

Comment:
And for which user it won't be significant? Why not to just enable it always? What concerns do you have around that?

Reply:
Well, I wasn't bold enough to argue with previous comment which stated `[...] should have message meaningful for users, for others (connection problem?) it's better to return generic message and log details...`

But I personally think that it is better to show error to the user - Seeing that you have similar though I've removed flag and set error always to be shown :slightly_smiling_face:

Reply:
Ok, now I understand. In which situation have you occurred other problem than `SQLSyntaxErrorException`?

Reply:
When you are using column which doesn't exists inside query - then in we would get e.g. `PSQLException`

Reply:
Uh, so it's look like we have to choose a compromise between technical information leakage like:
- problems with authentication
- connection problems 

that wasn't configured by a user and  helpfulness of the message.

Let's:
- add a comment that we made this informed choice
- adds a logs before returning message to make sure that also admin would have an access to these information

Reply:
You are right - thanks, I've added both comment and log entry :+1: